### PR TITLE
Added model and history as class attributes

### PR DESCRIPTION
**Pull request by jsalbert**
- I added both to class `Trainer` attributes to be able to access them after a training is done. 
It will allow to do more things with the model without having to load, and to extract metrics from the training without having to access tensorboard. 

- New Callback to follow min validation loss and renamed the callbacks weights filenames to give more information

### CODE REVIEW

File: keras_trainer/trainer.py [hunk 1]
```diff
@@ -71,6 +71,8 @@ def __init__(self, **options):
         if self.num_gpus is None:
             self.num_gpus = detect_num_gpus()
 
+        self.history = None
```
Comment (stephensolis) [inline keras_trainer/trainer.py]: You can leave this out

File: keras_trainer/trainer.py [hunk 2]
```diff
@@ -84,7 +86,7 @@ def __init__(self, **options):
 
     def run(self):
         # Set up the training data generator.
-        train_data_generator = self.train_generator or image.ImageDataGenerator(
+        train_data_generator = self.train_data_generator or image.ImageDataGenerator(
```
Comment (stephensolis) [inline keras_trainer/trainer.py]: Nice catch!
